From: Arnd Bergmann <arnd@arndb.de>
Subject: mips: spram: fix missing prototype warning for spram_config
Date: Mon, 4 Dec 2023 12:57:04 +0100

arch/mips/kernel/spram.c:194:6: error: no previous prototype for 'spram_config' [-Werror=missing-prototypes]

Link: https://lkml.kernel.org/r/20231204115710.2247097-15-arnd@kernel.org
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Cc: Stephen Rothwell <sfr@rothwell.id.au>
Cc: Thomas Bogendoerfer <tsbogend@alpha.franken.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/mips/include/asm/spram.h |    2 +-
 arch/mips/kernel/spram.c      |    1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

--- a/arch/mips/include/asm/spram.h~mips-spram-fix-missing-prototype-warning-for-spram_config
+++ a/arch/mips/include/asm/spram.h
@@ -3,7 +3,7 @@
 #define _MIPS_SPRAM_H
 
 #if defined(CONFIG_MIPS_SPRAM)
-extern __init void spram_config(void);
+extern void spram_config(void);
 #else
 static inline void spram_config(void) { }
 #endif /* CONFIG_MIPS_SPRAM */
--- a/arch/mips/kernel/spram.c~mips-spram-fix-missing-prototype-warning-for-spram_config
+++ a/arch/mips/kernel/spram.c
@@ -12,6 +12,7 @@
 #include <asm/mipsregs.h>
 #include <asm/r4kcache.h>
 #include <asm/hazards.h>
+#include <asm/spram.h>
 
 /*
  * These definitions are correct for the 24K/34K/74K SPRAM sample
_
